java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -360,13 +360,18 @@ public static void jdbcToArrowVectors(ResultSet rs, VectorSchemaRoot root, JdbcT
           root.getVector(rsmd.getColumnName(i)), config);
     }
 
-    try (CompositeJdbcConsumer compositeConsumer = new CompositeJdbcConsumer(consumers)) {
-      compositeConsumer.consume(rs);
-      root.setRowCount(compositeConsumer.getReadRowCount());
+    try (CompositeJdbcConsumer compositeConsumer =
```

Comment:
use the Iterator here?

Reply:
What is the reason? I think this is used for non-iterator API, do you mean it's special case of iterator API (with limit = -1 and just return the first schema root) ?

Reply:
For some reason, I would like to keep as it is here. For example
Users who use non-iterator API will get limited result which is not expected. (Since default targetBatchSize is not -1, if use iterator API here, it will only convert targetBatchSize data which is API breaking I think.)

Reply:
> What is the reason? I think this is used for non-iterator API, do you mean it's special case of iterator API (with limit = -1 and just return the first schema root) ?
Yes

I don't feel super strongly since we are deprecating these methods,  but the correct way of handling this would probably be to have a default value of something like "-2", then if we reach this location in the code and the value is -2 we replace it with -1, otherwise we honor the configured value.  Similarly we replace is it with some positive value inside the iterator.

Reply:
Thanks, have two questions:
1. should targetBatchSize also useful for non-iterator API? current is not
2. If question one is no, we could just replace targetBatchSize with -1 regarding less what it is.
If question one is yes, about the default value, you mean use -2 replace something like 1024 in JdbcToArrowConfig? I am a little confused here

Reply:
lets not touch this, we are keeping the functionality the same, and we can hopefully remove this code soon.